File: lab-3-ddb/environment/ruby_2.6.0/solution/query_table.rb
require "aws-sdk-dynamodb"

dynamodb_client = Aws::DynamoDB::Client.new(region: "us-east-1")

table_name = "lostcats"
name = "Puddles"

params = {
    table_name: table_name,
    expression_attribute_values: {
        ":p" => name,
    },
    key_condition_expression: "petname = :p",
}

begin
    result = dynamodb_client.query(params)
    puts "Query succeeded."
    result.items.each{|petname|
         puts "#{petname["notable_features"]}"
    }

rescue  Aws::DynamoDB::Errors::ServiceError => error
    puts "Unable to query table:"
    puts "#{error.message}"
end

File: lab-3-ddb/environment/python_3.6.8/query_table.py
import boto3
from boto3.dynamodb.conditions import Key, Attr

DDB_RESOURCE = boto3.resource("dynamodb", region_name="<FMI>")

petname = "Puddles"
table = DDB_RESOURCE.Table("lostcats")

response = <FMI>(
    KeyConditionExpression=Key("petname").eq(<FMI>),
    ProjectionExpression="<FMI>"
)

items = response['Items']
print(items)